src/L1/rollup/ScrollChain.sol

```diff
@@ -218,13 +319,75 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
         emit UpdateMaxNumTxInChunk(0, _maxNumTxInChunk);
     }
 
+    function initializeV2(uint128 _bundleSize) external reinitializer(2) {
+        // initialize tee proof state
+        uint256 cachedIndex = lastZkpVerifiedBatchIndex;
+        lastTeeVerifiedBatchIndex = cachedIndex;
+        emit VerifyBatchWithTee(
+            cachedIndex,
+            committedBatches[cachedIndex],
+            finalizedStateRoots[cachedIndex],
+            withdrawRoots[cachedIndex]
+        );
+
+        // initialize the first element in the array
+        bundleSize.push(BundleSizeStruct(_bundleSize, uint128(cachedIndex)));
```

Comment:
How will the upgrade happen? Basically right after executing the upgrade we'll need to reset all pending bundles in the rollup-relayer, right?

Reply:
Currently, the bundle size is about 100, right? We can use this for the rollup-relayer, and no need to do anything switch for rollup-relayer during upgrade.

Reply:
added relevant tweaks of bundle-proposer in this PR: https://github.com/scroll-tech/scroll/pull/1566